### PR TITLE
insert missing perl dependency for libbi 1.3.0

### PR DESCRIPTION
**Pull request by sbfnk**
### Have you:

- [X] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [X] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [X] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [X] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [X] Not altered the `bottle` section?
- [X] Checked that the tests still pass?

The changes are necessary for brewing to work with linuxbrew on older linux distributions and on travis-ci.

### CODE REVIEW

File: libbi.rb [hunk 1]
```diff
@@ -35,7 +35,12 @@ class Libbi < Formula
   depends_on "boost"
   depends_on "automake" => :run
 
-  resource "Getopt::ArgvFile" do
+  resource "Test::Simple" do
+    url "http://search.cpan.org/CPAN/authors/id/E/EX/EXODIST/Test-Simple-1.302075.tar.gz"
+    sha256 "86f2205498f96302e00331ac586bf366547e946e8637ad208d6317a2097d40b7"
+  end
+
+ resource "Getopt::ArgvFile" do
```
Comment (jonchang) [inline libbi.rb]: Indentation

File: libbi.rb [hunk 2]
```diff
@@ -102,10 +107,13 @@ class Libbi < Formula
 
   def install
     ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+    ENV.prepend_create_path "LD_LIBRARY_PATH", "#{Formula["netcdf"].lib}"
```
Comment (jonchang) [inline libbi.rb]: I think a bare `Formula["netcdf"].lib` should work, otherwise try `Formula["netcdf"].lib.to_s`

File: libbi.rb [hunk 3]
```diff
@@ -102,10 +107,13 @@ class Libbi < Formula
...
+    ENV.prepend_create_path "LD_LIBRARY_PATH", "#{Formula["netcdf"].lib}64"
+
     ENV.append "CPPFLAGS", "-I#{include}"
-    ENV.append "LDFLAGS", "-L#{Formula["qrupdate"].lib}"
+    ENV.append "LDFLAGS", "-L#{Formula["qrupdate"].lib} -L#{Formula["netcdf"].lib} -L#{Formula["netcdf"].lib}64"
 
-    perl_resources = [] << "Getopt::ArgvFile" << "Carp::Assert" << "File::Slurp" << "Parse::Yapp" << "Parse::Template" << "Parse::Lex" << "Parse::RecDescent" << "Math::Symbolic" << "Class::Inspector" << "File::ShareDir" << "Template" << "Graph"
+   perl_resources = [] << "Test::Simple" << "Getopt::ArgvFile" << "Carp::Assert" << "File::Slurp" << "Parse::Yapp" << "Parse::Template" << "Parse::Lex" << "Parse::RecDescent" << "Math::Symbolic" << "Class::Inspector" << "File::ShareDir" << "Template" << "Graph"
     include_resources = [] << "thrust"
```
Comment (jonchang) [inline libbi.rb]: These resources blocks are getting to be a bit of a mess. Could you try something like

```
resources.each do |r|
  r.stage do
    next if r.name == "thrust"
    # perl install stuff here
  end
end

resource("thrust").stage do |r| { ... }
```

Then you can get rid of the `perl_resources` and `include_resources` variables.

File: libbi.rb [hunk 4]
```diff
@@ -123,7 +131,7 @@ def install
       end
     end
 
-    system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+    system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}", "LDFLAGS="+ENV["LDFLAGS"]
```
Comment (jonchang) [inline libbi.rb]: Please use string interpolation here.

File: libbi.rb [hunk 5]
```diff
@@ -102,10 +107,13 @@ class Libbi < Formula
 
   def install
     ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+    ENV.prepend_create_path "LD_LIBRARY_PATH", "#{Formula["netcdf"].lib}"
+    ENV.prepend_create_path "LD_LIBRARY_PATH", "#{Formula["netcdf"].lib}64"
```
Comment (jonchang) [inline libbi.rb]: macOS doesn't have `lib64` folders so you'll need an `if OS.linux?` guard here:

```
==> Applying df6fbc815cc4c2c52f9a6bcbffc01bd82f9674fd.diff
/usr/bin/patch -g 0 -f -p1 -i /private/tmp/libbi--patch-20170201-91149-1quf5g1/df6fbc815cc4c2c52f9a6bcbffc01bd82f9674fd.diff
patching file lib/Bi/Builder.pm
Error: Operation not permitted - /usr/local/opt/netcdf/lib64
```

File: libbi.rb [hunk 6]
```diff
@@ -102,10 +107,13 @@ class Libbi < Formula
...
+
     ENV.append "CPPFLAGS", "-I#{include}"
-    ENV.append "LDFLAGS", "-L#{Formula["qrupdate"].lib}"
+    ENV.append "LDFLAGS", "-L#{Formula["qrupdate"].lib} -L#{Formula["netcdf"].lib} -L#{Formula["netcdf"].lib}64"
```
Comment (jonchang) [inline libbi.rb]: [See above](https://github.com/Homebrew/homebrew-science/pull/4970#discussion_r99009903)

**Comment by sbfnk** [comment]
All corrected in forced push. I also used the opportunity to add a flag to one of the perl modules that previously prevented successful installation on linux when brewing with `--verbose` (`TT_ACCEPT=y`).

File: libbi.rb [hunk 7]
```diff
@@ -1,9 +1,14 @@
 class Libbi < Formula
   desc "Bayesian state-space modelling on parallel computer hardware"
   homepage "http://libbi.org"
-  revision 1
+  revision 2
   head "https://github.com/libbi/LibBi.git"
 
+  patch do
```
Comment (jonchang) [inline libbi.rb]: Sorry I missed this before but could you please add a comment describing what this patch does, and file an upstream bug for it? We'd rather not maintain patches around if upstream can fix them.

**Comment by jonchang** [comment]
@sbfnk This is ready to merge, but could you please add a comment to the patch and file an upstream bug for it? I assume it is related to fixing builds on Linux but I am not certain

**Comment by sbfnk** [comment]
@jonchang Sorry, my bad! It was a testing patch for more verbose error logging that I had forgotten to take out. I've removed it now. Thanks for spotting.

**Comment by jonchang** [comment]
@BrewTestBot test this please

**Comment by jonchang** [comment]
Thanks @sbfnk! This has been merged.

**Pull request closed without merging**